kron.py

```diff
@@ -119,6 +120,19 @@ def _has_label(api_object: object, k: str, v: str) -> bool:
     return labels.get(k) == v
 
 
+def _api_exception_to_http_exception(e: ApiException):
+    """
+    Convert a Kubernetes ApiException into a HTTP exception.
+    """
+    log.error(e)
+    description = json.loads(e.body)["message"]
+    if e.status in [401, 403]:
+        return exceptions.Forbidden(description=description)
+    if e.status == 404:
+        return exceptions.NotFound(description=description)
+    return exceptions.InternalServerError("Something went wrong")
```

Comment:
Using "HTTP-aware" exceptions here breaks encapsulation, but I just don't think it is worth the effort to create an intermediate set of exceptions only to immediately convert them again.